src/Settings.elm

```diff
@@ -0,0 +1,132 @@
+module Settings exposing (..)
+
+import Accessibility.Aria as A11y exposing (..)
+import Html exposing (Html)
+import Html.Attributes as Attrs exposing (..)
+import Html.Events exposing (onClick)
+import Set exposing (Set)
+import SharedTypes exposing (..)
+
+
+viewSettingsForm : UserGroup -> Set String -> (String -> msg) -> Html msg
+viewSettingsForm userGroup isInputVisible onInputShow =
+    let
+        -- Extract settings data from the user group
+        settings =
+            userGroup.settingsData
+
+        -- Determine if the settings are inherited or not
+        isInherited =
+            case settings.inheritedFrom of
+                Just "Null" ->
+                    False
+
+                _ ->
+                    True
+
+        -- Helper function to check if a value is present (Just) or not (Nothing)
+        hasValue : Maybe Int -> Bool
+        hasValue value =
+            case value of
+                Just _ ->
+                    True
+
+                Nothing ->
+                    False
+
+        -- Extract retention policy from settings data
+        retentionPolicty =
+            settings.dataRetentionPolicy
+
+        -- Define a helper function to create input fields
+        inputfield : Maybe Int -> String -> Html msg
+        inputfield value fieldTitle =
+            let
+                -- Convert the Maybe Int value to a String
+                fieldValue =
+                    case value of
+                        Just int ->
+                            String.fromInt <| int
+
+                        Nothing ->
+                            "0"
+
+                -- Create a label for the input field
+                rowTitle =
+                    Html.label [ Attrs.class "block text-sm font-medium text-gray-700 mb-4" ] [ Html.text fieldTitle ]
+            in
+            -- Conditionally render an input field or a button based on whether the value is present
+            if hasValue value then
+                Html.div []
+                    [ rowTitle
+                    , Html.input
+                        [ Attrs.disabled (not isInherited)
+                        , Attrs.class "mt-1 p-2 block w-full border border-gray-300 rounded-md focus:ring focus:ring-blue-200 focus:border-blue-300"
+                        , Attrs.value fieldValue
+                        , A11y.label ("input for " ++ fieldTitle)
+                        ]
+                        []
+                    ]
+
+            else
+                Html.div []
+                    [ rowTitle
+                    , if Set.member fieldTitle isInputVisible then
+                        Html.input
```

Comment:
Q: Why did you choose plain text input? What is the advantage of it over a number one?

Reply:
input number would have been nice here too for the validation it provides